plugin/src/main/java/io/jenkins/plugins/remotingkafka/GlobalKafkaConfiguration.java

```diff
@@ -154,121 +194,77 @@ private void testConnection(String host, int port) throws IOException {
-        StandardListBoxModel result = new StandardListBoxModel();
-        if (item == null) {
-            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
-                return result.includeCurrentValue(credentialsId);
-            }
-        }
-        return result
-                .includeMatchingAs(
-                        ACL.SYSTEM,
-                        Jenkins.get(),
-                        StandardUsernamePasswordCredentials.class,
-                        Collections.singletonList(KAFKA_SCHEME),
-                        CredentialsMatchers.always()
-                )
-                .includeCurrentValue(credentialsId);
+        return fillCredentialsIdItems(item, credentialsId);
     }
 
     public FormValidation doCheckKafkaCredentialsId(@AncestorInPath Item item, @QueryParameter String value) {
-        if (item == null) {
-            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
-                return FormValidation.ok();
-            }
+        FormValidation checkedResult = checkCredentialsId(item, value);
+        boolean isAdminOrNullItem = (item != null || Jenkins.get().hasPermission(Jenkins.ADMINISTER));
+        if (isAdminOrNullItem && checkedResult.equals(FormValidation.ok())) {
+            this.kafkaCredentialsId = value;
         }
-        if (value.startsWith("${") && value.endsWith("}")) {
-            return FormValidation.warning("Cannot validate expression based credentials");
-        }
-        if (CredentialsProvider.listCredentials(
-                StandardUsernamePasswordCredentials.class,
-                Jenkins.get(),
-                ACL.SYSTEM,
-                Collections.singletonList(KAFKA_SCHEME),
-                CredentialsMatchers.withId(value)
-        ).isEmpty()) {
-            return FormValidation.error("Cannot find currently selected credentials");
-        }
-        this.kafkaCredentialsId = value;
-        return FormValidation.ok();
+        return checkedResult;
     }
 
+    @RequirePOST
     public ListBoxModel doFillSslTruststoreCredentialsIdItems(@AncestorInPath Item item, @QueryParameter String credentialsId) {
-        StandardListBoxModel result = new StandardListBoxModel();
-        if (item == null) {
-            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
-                return result.includeCurrentValue(credentialsId);
-            }
-        }
-        return result
-                .includeMatchingAs(
-                        ACL.SYSTEM,
-                        Jenkins.get(),
-                        StandardUsernamePasswordCredentials.class,
-                        Collections.singletonList(KAFKA_SCHEME),
-                        CredentialsMatchers.always()
-                )
-                .includeCurrentValue(credentialsId);
+        return fillCredentialsIdItems(item, credentialsId);
     }
 
     public FormValidation doCheckSslTruststoreCredentialsId(@AncestorInPath Item item, @QueryParameter String value) {
```

Comment:
`@RequirePOST` has not been applied to all methods. Was it intentional?

Reply:
@oleg-nenashev I got this error when putting `@RequirePost` for `doCheck` methods.
![screenshot from 2018-07-11 17-37-52](https://user-images.githubusercontent.com/13913977/42566562-2f305b40-8531-11e8-8a70-24087f354071.png)